src/findpeaks.jl

```diff
@@ -0,0 +1,131 @@
+module Findpeaks
+
+export findpeaks
+
+"""
+`findpeaks(y::Array{T},
+x::Array{S}=collect(1:length(y))
+;min_height::T=minimum(y), min_prom::T=minimum(y),
+min_dist::S=0, threshold::T=0 ) where {T<:Real,S}`\n
+Returns indices of local maxima (sorted from highest peaks to lowest)
+in 1D array of real numbers. Similar to MATLAB's findpeaks().\n
+*Arguments*:\n
+`y` -- data\n
+*Optional*:\n
+`x` -- x-data\n
+*Keyword*:\n
+`min_height` -- minimal peak height\n
+`min_prom` -- minimal peak prominence\n
+`min_dist` -- minimal peak distance (keeping highest peaks)\n
+`threshold` -- minimal difference (absolute value) between
+ peak and neighboring points\n
+"""
+function findpeaks(
+                   y :: AbstractVector{T},
+                   x :: AbstractVector{S} = collect(1:length(y))
+                   ;
+                   min_height :: T = minimum(y),
+                   min_prom :: T = zero(y[1]),
+                   min_dist :: S = zero(x[1]),
+                   threshold :: T = zero(y[1]),
```

Comment:
As for the first point, if we don't want to throw on empty data, this might be better:

```julia
function findpeaks(
                   y :: AbstractVector{T},
                   x :: AbstractVector{S} = collect(1:length(y))
                   ;
                   kwargs...
                  ) where {T <: Real, S}

    if isempty(y)
        return empty(x)
    end

    min_height = get(kwargs, :min_height, minimum(y))
    min_prom   = get(kwargs, :min_prom,   zero(y[1]))
    min_dist   = get(kwargs, :min_dist,   zero(x[1]))
    threshold  = get(kwargs, :threshold,  zero(y[1]))

    dy = diff(y)
...
```